src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -207,32 +207,25 @@ public override void Initialize(AnalysisContext context)
                 var idToAnalyzerMap = new ConcurrentDictionary<string, ConcurrentDictionary<string, ConcurrentBag<Location>>>();
                 compilationContext.RegisterOperationAction(operationAnalysisContext =>
                 {
-                    if (!(((IFieldInitializerOperation)operationAnalysisContext.Operation).Value is IObjectCreationOperation objectCreation))
+                    var fieldInitializer = (IFieldInitializerOperation)operationAnalysisContext.Operation;
```

Comment:
💭 Does this work for properties with this form? In theory the backing field initialization could be covered by one of these operations, but I'm not sure if it actually is.

```csharp
static DiagnosticDescriptor Rule { get; } = new DiagnosticDescriptor(...);
```

Reply:
Nope, that wasn't supported earlier either. I can file a separate feature request to support that case.

Reply:
Can you file this issue? I believe it's a common pattern for some third-party analyzers.